File: 백준/Silver/1629. 곱셈/곱셈.py
import sys

a, b, c = map(int, sys.stdin.readline().split())

def solution(a, b, c):
    if b == 0:
        return 1
    half = solution(a, b // 2, c)
    half = (half * half) % c
    if b % 2 == 0:
        return half
    else:
        return (half * a) % c

print(solution(a, b, c))

File: 백준/Silver/1629. 곱셈/곱셈.c
#include <stdio.h>

int solution(int a, int b, int c){
    if (b == 0){
        return 1;
    }

    long long half = solution(a, b / 2, c);
    half = (half * half) % c;

    if (b % 2 == 0){
        return half;
    } else {
        return (half * a) % c;
    }
}

int main(void){
    int a, b, c;
    scanf("%d %d %d", &a, &b, &c);
    int result = solution(a, b, c);
    printf("%d\n", result);

    return 0;
}